src/platform/site-wide/announcements/actions/index.js

```diff
@@ -32,8 +32,9 @@ export function initDismissedAnnouncements() {
   };
 }
 
-export function dismissAnnouncement(announcement) {
-  localAnnouncements(announcement);
+export function dismissAnnouncement(announcement, showEverytime) {
+  if (!showEverytime) localAnnouncements(announcement);
```

Comment:
Is there a better name for the `localAnnouncements` function? Or can we break it into two functions, since it seems to do two different things? Since we're separating the idea of dismissing the modal from whether it displays again, I think tweaking the function names to make things more clear is useful.